src/Raven.Server/ServerWide/Context/DocumentsOperationContext.cs

```diff
@@ -12,7 +12,18 @@ public class DocumentsOperationContext : TransactionOperationContext<DocumentsTr
 
         internal string LastDatabaseChangeVector;
         internal Dictionary<string, long> LastReplicationEtagFrom;
-        internal bool SkipChangeVectorValidation;
+        private bool _skipChangeVectorValidation;
+
+        internal bool SkipChangeVectorValidation
+        {
+            get => _skipChangeVectorValidation;
+            set
+            {
+                if (_skipChangeVectorValidation == false)
```

Comment:
So we _never_ switch from `true` to `false` ?

Reply:
Not in the same transaction.

Reply:
I don't follow. Can you elaborate? What's purpose of it then?

Reply:
This property is for debug assertion to validate that we do not moving the global change vector back in time (or conflicted). 

For example if it was `A:10 B:5` it shouldn't be set back to let's say `A:11` or `B:5`.
But there are some exceptions to this, for instance when we remove unused change vector entries. 
In this case we know that we will move the change vector back in time, so we need to skip the assertion.

We set the global change vector at the end of each transaction.
So once we set it to `true` we don't want to set it back to `false` until the transaction is over.

Reply:
Thanks. Makes sense